src/Host.cpp

```diff
@@ -683,49 +819,345 @@ void Host::send(QString cmd, bool wantPrint, bool dontExpandAliases)
+        }
+
+        return qMakePair(false, QStringLiteral("stopwatch with name \"%1\" (id:%2) was already stopped").arg(name, QString::number(watchId)));
+    }
+
+    // This should, indeed, be:
+    Q_UNREACHABLE();
+    return qMakePair(false, QStringLiteral("stopwatch with name \"%1\" (id:%2) not found").arg(name, QString::number(watchId)));
+}
+
+QPair<bool, QString> Host::stopStopWatch(const int id)
+{
+    auto pStopWatch = mStopWatchMap.value(id);
+    if (pStopWatch) {
+        if (pStopWatch->stop()) {
+            return qMakePair(true, QString());
+        }
+
+        return qMakePair(false, QStringLiteral("stopwatch with id %1 was already stopped").arg(id));
+    }
+
+    return qMakePair(false, QStringLiteral("stopwatch with id %1 not found").arg(id));
+}
+
+QPair<bool, QString> Host::resetStopWatch(const QString& name)
+{
+    auto watchId = findStopWatchId(name);
+    if (!watchId) {
+        if (name.isEmpty()) {
+            return qMakePair(false, QLatin1String("no unnamed stopwatches found"));
+        } else {
+            return qMakePair(false, QStringLiteral("stopwatch with name \"%1\" not found").arg(name));
+        }
+    }
+
+    auto pStopWatch = mStopWatchMap.value(watchId);
+    if (Q_LIKELY(pStopWatch)) {
+        if (pStopWatch->reset()) {
+            return qMakePair(true, QString());
+        }
+
+        if (name.isEmpty()) {
+            return qMakePair(false, QStringLiteral("the first unnamed stopwatch (id:%1) was already reset").arg(watchId));
+        } else {
+            return qMakePair(false, QStringLiteral("stopwatch with name \"%1\" (id:%2) was already reset").arg(name, QString::number(watchId)));
+        }
+    }
+
+    // This should, indeed, be:
+    Q_UNREACHABLE();
+    return qMakePair(false, QStringLiteral("stopwatch with name \"%1\" (id:%2) not found").arg(name, QString::number(watchId)));
+}
+
+QPair<bool, QString> Host::resetStopWatch(const int id)
+{
+    auto pStopWatch = mStopWatchMap.value(id);
+    if (pStopWatch) {
+        if (pStopWatch->reset()) {
+            return qMakePair(true, QString());
+        }
+
+        return qMakePair(false, QStringLiteral("stopwatch with id %1 was already reset").arg(id));
+    }
+
+    return qMakePair(false, QStringLiteral("stopwatch with id %1 not found").arg(id));
+}
+
+// Used when emulating past behavior for startStopWatch - there is only one
+// which takes a numeric argument as that is what old scripts will be using and
+// not a text name:
+QPair<bool, QString> Host::resetAndRestartStopWatch(const int id)
+{
+    auto pStopWatch = mStopWatchMap.value(id);
+    if (pStopWatch) {
+        pStopWatch->stop();
+        pStopWatch->reset();
+        pStopWatch->start();
+        return qMakePair(true, QString());
     }
+
+    return qMakePair(false, QStringLiteral("stopwatch with id %1 not found").arg(id));
 }
 
-bool Host::resetStopWatch(int watchID)
+bool Host::makeStopWatchPersistent(const int id, const bool state)
```

Comment:
Why is there no name overload of this function?

Reply:
Because looking up the name is done in the `TLuaInterpreter` code and that handles producing the `not found` message which is the only failure mechanism (so no need to have to pass back an error message)...